docs/dptp-triage-sop/azure-credentials-rotation.md

```diff
@@ -0,0 +1,12 @@
+# Azure Credentials Rotation
+
+Because [Azure forces an expiry date for the credentials](https://redhat-internal.slack.com/archives/CBN38N3MW/p1675693067516879?thread_ts=1675591291.220619&cid=CBN38N3MW), we have to refresh them in our secrets.
+
+They are stored in the items `os4-installer.openshift-ci.azure` and `os4-installer.openshift-ci2.azure` in Vault.
+To rotate the credentials,
+
+- Ask [#forum-dp-platform](https://redhat-internal.slack.com/archives/CBUT43E94) for a new `Secret`.
+- Replace the value of `clientSecret` of `osServicePrincipal.json` in the above TWO items in Vault with the SAME `Secret` obtained from the DPP team.
+Note that the two sets of credentials are from the same Service Principal, hence the same `Secret`, with different subscriptions.
```

Comment:
Since these are fully under the control of DPP, couldn't we just move them to [user credentials](https://docs.ci.openshift.org/docs/how-tos/adding-a-cluster-profile#providing-credentials), removing ourselves from the rotation process?

Reply:
Before I ask DPP if they want to take the ownership of the secret, I think the blocker here is that the secret e.g. [cluster-secrets-azure4](https://github.com/openshift/release/blob/11767dd6283dd48d19a937ab0e048270964df752/core-services/ci-secret-bootstrap/_config.yaml#L1784-L1823) includes more than `osServicePrincipal.json`. The rest of it is still managed by us. Do we support manage a secret whose data are partially from self-service and partially from dptp?

Reply:
Yes, that's how we compose cluster profiles, as described in that page.  E.g.:

```json
{
  "openvpn.conf": "…",
  "openvpn.sh": "…",
  "secretsync/target-name": "cluster-secrets-nutanix",
  "secretsync/target-namespace": "ci",
  "vpn.yaml": "…",
}
```
https://vault.ci.openshift.org/ui/vault/secrets/kv/show/selfservice/cluster-secrets-nutanix/cluster-profile

```console
$ yq -r '.secret_configs[]|select(.to[0].name=="cluster-secrets-nutanix").from|keys[]' core-services/ci-secret-bootstrap/_config.yaml | sort
.awscred
insights-live.yaml
pull-secret
ssh-privatekey
ssh-publickey
```

```console
$ oc --context build01 --namespace ci get secret cluster-secrets-nutanix -o json | jq -r '.data|keys[]' | sort
.awscred
insights-live.yaml
openvpn.conf
openvpn.sh
pull-secret
secretsync-vault-source-path
ssh-privatekey
ssh-publickey
vpn.yaml
$ oc --context build01 --namespace ci get secret cluster-secrets-nutanix -o json | jq -r '.data["secretsync-vault-source-path"]' | base64 -d; echo
kv/selfservice/cluster-secrets-nutanix/cluster-profile
```

Reply:
Asking DPP.
https://redhat-internal.slack.com/archives/GB7NB0CUC/p1677099658389789